### ISSUE TITLE
$HOSTNAME in vhosts.d/portus.conf

### ISSUE THREAD

**Issue by DirkHeinke** [SUSE/Portus#358]
It looks like $HOSTNAME is not set in [this](https://github.com/SUSE/Portus/blob/master/packaging/suse/scripts/configure_apache.sh#L17) script? Something leads to an apache configuration file with this content:
```
[....]
SetEnv PORTUS_SECRET_KEY_BASE XXXXXXXXXXXXXXXXX
SetEnv PORTUS_MACHINE_FQDN ${HOSTNAME}
SetEnv PORTUS_KEY_PATH /srv/Portus/config/server.key
[...]
```
I guess this env is read by the application [here](https://github.com/SUSE/Portus/blob/master/config/secrets.yml#L34). So that the issued token contains:
```
{
  "iss": "${HOSTNAME}",
  "sub": "dirk",
[...]
```
and lets the `docker login` command fail:
```
Error response from daemon: no successful auth challenge for https://portus.xxx:5000/v2/ - errors: [token auth attempt to https://portus.xxx:5000/v2/ realm "https://portus.xxx/v2/token" failed with status: 401 Unauthorized]
```

Docker registry log:
```
2015-09-22T09:03:30.829112+00:00 portus registry[1108]: time="2015-09-22T09:03:30Z" level=error msg="token from untrusted issuer: \"${HOSTNAME}\""
```

**Comment by jordimassaguerpla** [comment]
Hi Dirk! Thanks for the report. We have already fixed that in 

https://github.com/SUSE/Portus/commit/effdebdc6f9193a46b3b0a5644fbac6daf3b66ba

I guess your appliance is not the latest one.

Could you send us the output of

rpm -qi Portus

This will tell us which is the latest commit.

Thanks again!

**Comment by jordimassaguerpla** [comment]
actually the appliance does not call setup_apache and that is why this is not working

**Comment by DirkHeinke** [comment]
Hi,
here is the output:
```
rpm -qi Portus
Name        : Portus
Version     : 20150918120318
Release     : 1.1
Architecture: x86_64
Install Date: Fri Sep 18 17:29:10 2015
Group       : System/Management
Size        : 53624951
License     : Apache-2.0
Signature   : RSA/SHA1, Fri Sep 18 12:37:38 2015, Key ID a193fbb572174fc2
Source RPM  : Portus-20150918120318-1.1.src.rpm
Build Date  : Fri Sep 18 12:36:56 2015
Build Host  : build85
Relocations : (not relocatable)
Vendor      : obs://build.opensuse.org/Virtualization
URL         : https://github.com/SUSE/Portus
Summary     : Authorization service and fronted for Docker registry (v2)
Description :
Portus targets version 2 of the Docker registry API. It aims to act both as an authoritzation server and as a user interface for the next generation of the Docker registry.

This package has been built with commit 75ce9316e3f126b49d2c9948030eda99c98acb49 from branch master on date Fri, 18 Sep 2015 12:03:18 +0000
Distribution: Virtualization:containers:Portus / openSUSE_13.2
```

Looks like this is newer than the commit, but I'll try the newest build (mine is Build62.6, I'll use 62.9)

**Comment by jordimassaguerpla** [comment]
I just created a PR that fixes this issue

https://github.com/SUSE/Portus/pull/360

**Comment by jordimassaguerpla** [comment]
also the appliance 

https://build.opensuse.org/package/rdiff/Virtualization:containers:Portus/PortusAppliance?linkrev=base&rev=63

**Comment by mssola** [comment]
This has been fixed in commit 9c32ed9840a2f04d20c856295864b43289402d7f. Basically, this is now handled by the new `portusctl` tool. Feel free to re-open this issue.
